coinmetrics-bytecoin/CoinMetrics/Bytecoin.hs

```diff
@@ -0,0 +1,193 @@
+{-# LANGUAGE DeriveGeneric, OverloadedLists, OverloadedStrings, StandaloneDeriving, TemplateHaskell, TypeFamilies, ViewPatterns #-}
+
+module CoinMetrics.Bytecoin
+  ( Bytecoin(..)
+  , BytecoinBlock(..)
+  , BytecoinTransaction(..)
+  , BytecoinTransactionInput(..)
+  , BytecoinTransactionOutput(..)
+  ) where
+
+import qualified Data.Aeson as J
+import qualified Data.Aeson.Types as J
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.ByteString.Short as BS
+import qualified Data.Text.Encoding as T
+import qualified Data.HashMap.Lazy as HML
+import qualified Data.Vector as V
+import Data.Time.Clock.POSIX
+import Data.Maybe
+import Data.Int
+import Data.Proxy
+import Hanalytics.Schema
+
+import CoinMetrics.BlockChain
+import CoinMetrics.JsonRpc
+import CoinMetrics.Schema.Flatten
+import CoinMetrics.Schema.Util
+import CoinMetrics.Util
+
+newtype Bytecoin = Bytecoin JsonRpc
+
+data BytecoinBlock = BytecoinBlock
+  { bc_difficulty :: {-# UNPACK #-} !Int64
+  , bc_hash :: {-# UNPACK #-} !HexString
+  , bc_height :: {-# UNPACK #-} !Int64
+  , bc_major_version :: {-# UNPACK #-} !Int64
+  , bc_minor_version :: {-# UNPACK #-} !Int64
+  , bc_nonce :: {-# UNPACK #-} !Int64
+  , bc_reward :: {-# UNPACK #-} !Int64
+  , bc_size :: {-# UNPACK #-} !Int64
+  , bc_timestamp :: {-# UNPACK #-} !Int64
+  , bc_coinbase_transaction :: !BytecoinTransaction
+  , bc_transactions :: !(V.Vector BytecoinTransaction)
+  }
+
+instance IsBlock BytecoinBlock where
+  getBlockHeight = bc_height
+  getBlockTimestamp = posixSecondsToUTCTime . fromIntegral . bc_timestamp
+
+newtype BytecoinBlockWrapper = BytecoinBlockWrapper
+  { unwrapBytecoinBlock :: BytecoinBlock
+  }
+
+instance J.FromJSON BytecoinBlockWrapper where
+  parseJSON = J.withObject "bytecoin block" $ \fields -> fmap BytecoinBlockWrapper $ BytecoinBlock
+    <$> (fields J..: "difficulty")
+    <*> (fields J..: "hash")
+    <*> (fields J..: "height")
+    <*> (fields J..: "major_version")
+    <*> (fields J..: "minor_version")
+    <*> (fields J..: "nonce")
+    <*> (fields J..: "reward")
+    <*> (fields J..: "size")
+    <*> (fields J..: "timestamp")
+    <*> (unwrapBytecoinTransaction <$> fields J..: "coinbase_transaction")
+    <*> (V.map unwrapBytecoinTransaction <$> fields J..: "transactions")
+
+data BytecoinTransaction = BytecoinTransaction
+  { mt_extra :: {-# UNPACK #-} !HexString
```

Comment:
Here `mt_` prefix is apparently copied from Monero, meaning Monero transaction. Probably `bt_` is better.